### PR TITLE
feat: explore page [6.4]

### PR DESCRIPTION
**Pull request by nuanyang233**
# Pull Request

## Related Issue

Closes [FW-2687](https://mask.atlassian.net/browse/FW-2687) FW-2702

## Checklist

-   [ ] I have tested these changes locally.
-   [ ] I have reviewed the code changes.
-   [ ] I have added unit tests if applicable.

[FW-2687]: https://mask.atlassian.net/browse/FW-2687?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/(normal)/discover/pages/Discover.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import { DiscoverPostList } from '@/components/Posts/DiscoverPostList.js';
+import { DiscoverSnapshotList } from '@/components/Snapshot/DiscoverSnapshotList.js';
+import { type DiscoverSource, Source } from '@/constants/enum.js';
+
+interface Props {
+    source: DiscoverSource;
+}
+
+export function DiscoverPage({ source }: Props) {
+    if (source === Source.Snapshot) {
```
Comment (guanbinrui) [inline src/app/(normal)/discover/pages/Discover.tsx:16]: safeUnreachable

File: src/components/ProfileInList.tsx [hunk 2]
```diff
@@ -51,17 +59,38 @@ export function ProfileInList({ profile, noFollowButton, listKey, index }: Profi
...
+                                className={classNames('gap-1 leading-[22px] hover:underline', {
+                                    'pointer-events-none': source !== Source.Farcaster && source !== Source.Lens,
+                                })}
                             >
-                                @{profile.handle}
+                                <data value={followerCount}>
+                                    <span className="font-bold text-lightMain">{nFormatter(followerCount)} </span>
+                                    <span className="text-secondary">
+                                        <Plural value={followerCount} one="Follower" other="Followers" />
+                                    </span>
```
Comment (guanbinrui) [inline src/components/ProfileInList.tsx:90]: <Plural />

File: src/components/ChannelInList.tsx [hunk 3]
```diff
@@ -81,21 +81,23 @@ export function ChannelInList({
...
-                            height={isSmall || dense ? 14 : 18}
-                            className="shrink-0"
-                        />
-                        <data value={channel.followerCount}>{nFormatter(channel.followerCount)}</data>
+                        <span className="mx-1 leading-5 text-secondary">·</span>
+                        <data value={channel.followerCount}>
+                            <span className="font-bold text-lightMain">{nFormatter(channel.followerCount)} </span>
+                            <span className="text-secondary">
+                                <Plural value={channel.followerCount} one="Follower" other="Followers" />
+                            </span>
```
Comment (guanbinrui) [inline src/components/ChannelInList.tsx:99]: <Plural />

File: src/components/TokenTrendingList.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            switch (type) {
+                case TrendingType.TopGainers:
+                case TrendingType.TopLosers:
+                    return Coingecko.getTopGainersOrLosers(type);
+                case TrendingType.Trending:
+                    return Coingecko.getTopTrendingCoins();
+                case TrendingType.Meme:
+                    return Coingecko.getTopMemeCoins();
+                default:
+                    return [] as TokenWithMarket[];
```
Comment (guanbinrui) [inline src/components/TokenTrendingList.tsx:23]: safeUnreachable

File: src/helpers/resolveTrendingName.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { TrendingType } from '@/constants/enum.js';
+import { UnreachableError } from '@/constants/error.js';
+import { createLookupTableResolver } from '@/helpers/createLookupTableResolver.js';
+
+export const resolveTrendingName = createLookupTableResolver<TrendingType, string>(
```
Comment (guanbinrui) [inline src/helpers/resolveTrendingName.ts:5]: It is not used anywhere.

File: src/constants/enum.ts [hunk 6]
```diff
@@ -96,6 +99,19 @@ export enum DiscoverType {
     TopChannels = 'top-channels',
 }
 
+export enum ExploreType {
+    CryptoTrends = 'crypto-trends',
+    TopProfiles = 'top-profiles',
+    TopChannels = 'top-channels',
+}
+
+export enum TrendingType {
```
Comment (guanbinrui) [inline src/constants/enum.ts:108]:
```suggestion
export enum CryptoTrendingType {
```

File: src/app/(normal)/explore/pages/Explore.tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+interface Props {
+    source: ExploreSource;
+    type: ExploreType;
+}
+
+export function ExplorePage({ source, type }: Props) {
+    if (type === ExploreType.CryptoTrends) {
+        return <TokenTrendingList type={source as TrendingType} />;
+    } else {
+        return <ExploreContentList source={source as SocialSource} type={type} />;
```
Comment (guanbinrui) [inline src/app/(normal)/explore/pages/Explore.tsx:20]: safeUnreachable

File: src/app/(normal)/explore/pages/Explore.tsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
+'use client';
+
+import { safeUnreachable } from '@masknet/kit';
+import { type ComponentType, memo } from 'react';
+
+import { ChannelList } from '@/components/Channel/ChannelList.js';
+import SuggestedFollowUsersList from '@/components/SuggestedFollows/SuggestedFollowUsersList.js';
+import { TokenTrendingList } from '@/components/TokenTrendingList.js';
```
Comment (guanbinrui) [inline src/app/(normal)/explore/pages/Explore.tsx:8]: Move this to this file?

File: src/app/(normal)/explore/pages/Explore.tsx [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+export function ExplorePage({ source, type }: Props) {
+    if (type === ExploreType.CryptoTrends) {
+        return <TokenTrendingList type={source as TrendingType} />;
+    } else {
+        return <ExploreContentList source={source as SocialSource} type={type} />;
+    }
+}
+
+export const ExploreContentList: ComponentType<{ type: ExploreType; source: SocialSource }> = memo(
```
Comment (guanbinrui) [inline src/app/(normal)/explore/pages/Explore.tsx:24]: Merge into `ExplorePage`.

File: src/components/ProfileInList.tsx (hunk 2 above)
Comment (nuanyang233) [reply src/components/ProfileInList.tsx:90]: It's just migrated from other codes, the styles of count number and plural are different.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**